opm/grid/cpgrid/CpGridData.cpp

```diff
@@ -2509,6 +2606,65 @@ CpGridData::refinePatch(const std::array<int,3>& cells_per_dim, const std::array
             parent_to_children_cells, child_to_parent_faces, child_to_parent_cells};
 }
 
+bool CpGridData::mark(int refCount, const cpgrid::Entity<0>& element)
+{
+    if (refCount == -1) {
+        OPM_THROW(std::logic_error, "Coarsening is not supported yet.");
+    }
+    // Check the cell to be marked for refinement has no NNC (no neighbouring connections). Throw otherwise.
+    if (hasNNCs({element.index()}) && (refCount == 1)) {
+        OPM_THROW(std::logic_error, "NNC face on a cell containing LGR is not supported yet.");
```

Comment:
Maybe use message "Refinement of cells with face representing an NNC is not supported, yet"?